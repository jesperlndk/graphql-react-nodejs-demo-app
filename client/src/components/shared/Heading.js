import React from 'react';
import injectSheet from 'react-jss';
import classNames from 'classnames';

const headingTags = ['h1', 'h2', 'h6'];
const getHeadingTag = type => headingTags[headingTags.indexOf(type)] || headingTags[0];

const Heading = ({ text, sub, type, classes, className }) => {
  const Tag = getHeadingTag(type);
  return (
    <Tag className={classNames(classes.title, classes[Tag], className)}>
      {text}
      {sub ? (<span className={classNames(classes.sub)}>{sub}</span>) : null}
    </Tag>
  );
};

const styles = theme => ({
  title: {
    margin: 0,
    padding: 0,
    fontFamily: theme.fonts.heading,
  },
  sub: {
    display: 'block',
  },

  h1: {
    fontSize: 40,
    fontWeight: theme.weights.medium,
    lineHeight: 1.4,
    color: theme.colors.primary,

    ...theme.responsive.desktop({
      fontSize: 36,
    }),
    ...theme.responsive.tablet({
      fontSize: 32,
    }),
    ...theme.responsive.mobile({
      fontSize: 26,
    }),

    '& span': {
      fontSize: 36,
      fontWeight: theme.weights.light,

      ...theme.responsive.desktop({
        fontSize: 33,
      }),
      ...theme.responsive.tablet({
        fontSize: 30,
      }),
      ...theme.responsive.mobile({
        fontSize: 24,
      }),
    },
  },

  h2: {
    fontWeight: theme.weights.light,
    fontSize: 24,
    color: theme.colors.secondary,
    lineHeight: 1.35,

    ...theme.responsive.tablet({
      fontSize: 22,
    }),
    ...theme.responsive.mobile({
      fontSize: 19,
    }),

    '& span': {
      fontWeight: theme.weights.medium,
      fontSize: 36,
      color: theme.colors.primary,

      ...theme.responsive.tablet({
        fontSize: 30,
      }),
      ...theme.responsive.mobile({
        fontSize: 24,
      }),
    },
  },

  h6: {
    fontSize: 14,
    fontWeight: theme.weights.bold,
    textTransform: 'uppercase',
    letterSpacing: 1,
  },
});

export default injectSheet(styles)(Heading);
