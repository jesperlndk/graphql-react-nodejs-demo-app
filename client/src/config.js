const {
  NODE_ENV,
  REACT_APP_GOOGLE_API_KEY,
  REACT_APP_GRAPHQL_URL,
} = process.env;

const isProduction = NODE_ENV === 'production';

const dev = {
  graphql: {
    url: 'http://localhost:4000/',
  },
};

const prod = {
  graphql: {
    url: REACT_APP_GRAPHQL_URL,
  },
};

export default {
  ...isProduction ? prod : dev,
  /* general: */
  theme: 'standard',
  general: {
    appName: 'Movie Explorer',
    city: 'San Francisco',
  },
  images: {
    remoteUrlBase: 'https://image.tmdb.org/t/p/',
  },
  googleMaps: {
    apiKey: REACT_APP_GOOGLE_API_KEY,
  },
};
