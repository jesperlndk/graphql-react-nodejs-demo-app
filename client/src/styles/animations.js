export default {
  fadeIn: (delay = '0s', duration = '2s') => ({
    opacity: 0,
    animationDelay: delay,
    animation: `fadeIn-animation ${duration} normal forwards`,
  }),
  fadeInSlideUp: (delay = '0s', duration = '2s') => ({
    opacity: 0,
    animationDelay: delay,
    animation: `fadeInSlideUp-animation ${duration} cubic-bezier(0, 1, 0.5, 1) 1 normal forwards`,
  }),
  fadeInSlideUpSubtle: (delay = '0s', duration = '2s') => ({
    opacity: 0,
    animationDelay: delay,
    animation: `fadeInSlideUpSubtle-animation ${duration} cubic-bezier(0, 1, 0.5, 1) 1 normal forwards`,
  }),
};
