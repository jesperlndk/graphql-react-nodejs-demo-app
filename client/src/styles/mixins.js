export default {
  backgroundImageCover: (bgImage = '', position = 'top center') => ({
    backgroundImage: `url(${bgImage})`,
    backgroundPosition: position,
    backgroundSize: 'cover',
  }),
};
