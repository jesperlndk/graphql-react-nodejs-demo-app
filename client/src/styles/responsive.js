const responsiveStyle = (maxWidth, style = {}) => ({
  [`@media (max-width: ${maxWidth}px)`]: {
    ...style,
  },
});

export default {
  desktop: (style = {}) => responsiveStyle(1100, style),
  tablet: (style = {}) => responsiveStyle(900, style),
  mobile: (style = {}) => responsiveStyle(650, style),
};
