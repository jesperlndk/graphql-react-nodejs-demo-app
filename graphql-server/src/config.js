const {
  NODE_ENV,
  PORT,
  // env variables:
  API_KEY_TMDB,
  MOVIES_API_URL,
} = process.env;

const dev = {
  playground: true,
  port: 4000,
  apis: {
    movies: 'http://localhost:5000/movies',
    tmdb: `https://api.themoviedb.org/3/$PATH?api_key=${API_KEY_TMDB}&language=en-US&`,
  },
};

const prod = {
  playground: false,
  port: PORT,
  apis: {
    movies: MOVIES_API_URL,
    tmdb: `https://api.themoviedb.org/3/$PATH?api_key=${API_KEY_TMDB}&language=en-US&`,
  },
};

const isDev = NODE_ENV !== 'production';

module.exports = {
  ...isDev ? dev : prod,
  /* general: */
  path: '/',
  personTypeMap: {
    actor: ['actor_1', 'actor_2', 'actor_3'],
    writer: ['writer'],
    director: ['director'],
  },
  personNamesIgnoreList: ['N/A'],
};
