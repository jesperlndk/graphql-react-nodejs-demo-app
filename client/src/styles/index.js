// css
import './css/fonts.css';
import './css/normalize.css';
import './css/animations.css';

// js
import animations from './animations';
import mixins from './mixins';
import responsive from './responsive';

// themes
import standard from './themes/standard';

const themes = { standard };

const styles = theme => ({
  ...themes[theme],
  animations,
  mixins,
  responsive,
});

export default styles;
