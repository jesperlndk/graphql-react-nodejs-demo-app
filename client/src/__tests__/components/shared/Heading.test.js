import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'react-jss';
import styles from '../../../styles';
import config from '../../../config';
import Heading from '../../../components/shared/Heading';

describe('Heading component', () => {
  it('should render empty h1 tag by default', () => {
    const render = (
      <ThemeProvider theme={styles(config.theme)}>
        <Heading />
      </ThemeProvider>
    );
    const tree = renderer.create(render).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render text and sub-text inside if given', () => {
    const render = (
      <ThemeProvider theme={styles(config.theme)}>
        <Heading text="My headline" sub="My sub-headline" />
      </ThemeProvider>
    );
    const tree = renderer.create(render).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should add class to heading tag if any className given', () => {
    const render = (
      <ThemeProvider theme={styles(config.theme)}>
        <Heading type="h2" text="My headline" className="my-fancy-custom-class" />
      </ThemeProvider>
    );
    const tree = renderer.create(render).toJSON();
    expect(tree.props.className).toContain('my-fancy-custom-class');
  });
});
