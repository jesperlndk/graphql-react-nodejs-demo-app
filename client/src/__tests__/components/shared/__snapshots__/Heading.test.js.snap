// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should render empty h1 tag by default 1`] = `
<h1
  className="Heading-title-0-1-1 Heading-h1-0-1-3"
/>
`;

exports[`Heading component should render text and sub-text inside if given 1`] = `
<h1
  className="Heading-title-0-1-6 Heading-h1-0-1-8"
>
  My headline
  <span
    className="Heading-sub-0-1-7"
  >
    My sub-headline
  </span>
</h1>
`;
